utils/utils.go

```diff
@@ -591,6 +591,13 @@ func CustomizePodSpec(pts *corev1.PodTemplateSpec, ba common.BaseComponent) {
 		})
 	}
 
+	// This ensures that the pods are updated if the service account is updated
```

Comment:
The pods should be updated for user-provided service account as well. Setting the reference value at the end of `ServiceAccountPullSecretExists` would ensure that the right service account is checked for existence, it has image pull secret (if it exists) and finally set the resource version of the corresponding service account.